clippy_lints/src/coerce_any_ref_to_any.rs

```diff
@@ -0,0 +1,108 @@
+use clippy_utils::diagnostics::span_lint_and_sugg;
+use clippy_utils::source::snippet;
+use clippy_utils::sym;
+use rustc_errors::Applicability;
+use rustc_hir::{Expr, ExprKind};
+use rustc_lint::{LateContext, LateLintPass};
+use rustc_middle::ty::{self, ExistentialPredicate, Ty, TyCtxt};
+use rustc_session::declare_lint_pass;
+
+declare_clippy_lint! {
+    /// ### What it does
+    ///
+    /// Protects against unintended coercion of references to container types to `&dyn Any`
+    /// when the container type dereferences to a `dyn Any` which could be reborrowed instead.
```

Comment:
I think you're using `reborrowed` as a synonym to `rereference`, which is valid, but in this case, it might be better not invoke Borrowing in the user's mind who wouldn't have context on this lint, since this lint has very little to do with the borrow checker.

Perhaps:

```
when the container type dereferences to a `dyn Any` which could be directly referenced instead.
```